File: 21.11.01_ram.py
# -*- coding: utf-8 -*-
# UTF-8 encoding when using korean
import math
in1 = int(input())
while not 1<=in1<=100:
	in1 = int(input())
in2 = input()
in3 = in2.split(" ")
in4 = {}
for i in range(len(in3)):
	in4[i] = int(in3[i])
	while not 1<=len(in4)<=in1 or not 100<=in4[i]<=math.pow(2, 31)-1:
		in2 = input()
sum = in1
result = ""
for i in range(len(in4)):
	result = result + str(i+1) + " "
	for j in range(32):
		if(in4[i] == math.pow(2, j)):
			sum = sum -1
			result = result.replace(str(i+1) + " ", "")
			
print(sum)
print(result)

File: 21.11.01_ram.java
import java.io.*;
import java.util.*;
class Main {
	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		sc.nextLine();
		String m = sc.nextLine();
		String[] m2 = m.split(" ");
		int[] m3 = new int[m2.length];
		int sum = n;
		String result = "";
		for (int i = 0; i < m2.length; i++){
			m3[i] = Integer.parseInt(m2[i]);
			boolean t = true;
			result = result + Integer.toString(i+1) + " ";
			for(int j =1; j<31; j++){
				if(m3[i]==Math.pow(2,j)){
					sum--;
					result = result.replace(Integer.toString(i+1) + " ", "");
					
				}	
			}
		}
		System.out.println(sum);
		System.out.println(result);
		
	}
}
